### DISCUSSION TITLE
Move to dedicated organization

### DISCUSSION THREAD

**Discussion by pftg** [donv/capybara-screenshot-diff#115] [General]
Hey @donv, can we set up a separate organization to add more services? I'll be more flexible with the setup tools and add more guys to it so that we can increase awareness and our capacity.

**Comment by UweKubosch** [comment]
Sounds good.

**Comment by donv** [comment]
Yes, I'll get on it.

**Comment by UweKubosch** [comment]
OK, moved the repo to a new organization.

**Comment by pftg** [comment]
@UweKubosch about name options:

0. capybara-screenshot-diff
Rationale: Pretty straightforward

2. SnapDiff (snap_diff)
Rationale: "Snap" refers to screenshots, and "Diff" highlights the comparison functionality. It's short, easy to remember, and directly hints at the core functionality of the project.

3. CapyDiff (capy_diff)
Rationale: This name combines "Capybara" with "Diff," maintaining a connection to the original gem while emphasizing the difference detection feature. It's unique, catchy, and related to the project's context.

4. ScreenCompare
Rationale: This name clearly describes the main purpose of the gem – comparing screenshots. It's straightforward, easy to pronounce, and conveys the project's functionality at a glance.

I'd vote for SnapDiff (snap-diff) or any variations like (snapdiff), or have the same capybara-screenshot-diff

**Comment by donv** [comment]
I like SnapDiff, but if the thought is to have multiple gems in the same organization, the organization name should be more general than each gem, right?  snap_diff  would be the name of one gem, right?

**Reply by pftg** [reply]
Yeah, you are right.

Maybe like `percy` do: `snap_diff`, `snap_diff-capybara`

**Reply by pftg** [reply]
oh, and I have extra html reporter as plugin to add to help to review changed files, I already used it on my project.

**Reply by donv** [reply]
Yeah, sounds good.  I'll rename.

**Reply by donv** [reply]
Like that?  The organization name is `snap-diff` since it looks good and underscores are not allowed.  The gem name will be `snap_diff-capybara` with name spaces `SnapDiff` and `SnapDiff::Capybara`, etc.

The migration path from `capybara-screenshot-diff` gem should be as smooth as possible with `capybara-screenshot-diff` requiring and using `snap_diff-capybara` for the first major version, similar to `sass` using `sassc`.

**Reply by donv** [reply]
Can this gem be used without capybara?

**Reply by pftg** [reply]
@donv I do not have any clue about what is the best call ;). I'd not rush with conclusions for now.

My Vision: to deliver a free competitor to Percy without requiring a standalone server.

So far, my motivation: 

* extract minimalistic image comparison lib (for guys to compare in tests pdf documents): `SnapDiff.matched?(a_file, b_file)`
* deliver several reporters to quickly overview changes without extra steps like opening the editor.
  * need smooth integration with GitHub and GitLab to simplify the headache of reviewing them
* easy to run screenshots for online sites or any other servers (ex: Hugo) without coding much. `bin/snap_diff links.yml`

Some thoughts: 

1. Would we deliver other frameworks like Cypress and Nightwatch? Low chances.
2. Capybara is a wrapper around web drivers. We should eventually support any web driver without Capybara. `SnapDiff.snapshot(driver, 'name')` or `SnapDiff.snapshot_matched?(driver, 'name')`

**Reply by UweKubosch** [reply]
Sounds good.  👍

**Comment by pftg** [comment]
@UweKubosch @donv please give me a little more rights for this org ;) that I can setup [coderabbit](https://coderabbit.ai/) for auto code review

**Reply by donv** [reply]
What rights do you need?

**Reply by donv** [reply]
OK, I made you an owner.  Use it responsibly 😄
